### ISSUE TITLE
Print nodes in registration cache

### ISSUE THREAD

**Issue by paladincorners** [juanfont/headscale#1608]
## Why

Currently in order to register a new node you have to manually type in a long node key, where it is easy to miss a character. Copy / paste may not be always possible. If you could print the nodes and their node keys in the registration cache, you could simply find the right node key and register it from the CLI. Additionally, this has a security benefit where you could check outstanding registration requests on your network and monitor if there are any unusual unauthorized attempts.

## Description

Add a feature to print the node registration requests in the registration cache. Similar to how you have `headscale nodes list` you could do something like:
```
headscale regcache list
```
Or:
```
headscale regcache print
```

**Comment by vegardx** [comment]
Have you looked at `headscale preauthkeys`?

**Comment by paladincorners** [comment]
Hi! Yes I have, however for the authentication aspect, preauthkeys:
1. Require using the terminal to register the node on a computer (on a Mac for example you can now register a node with Headscale using only the GUI)
2. Are not supported on iOS at all
3. Still have the same issue, but in reverse - getting the preauthkey to your end user and explaining to them how to use it

Aditionally, there is still the benefit of monitoring registration attempts to your instance.

**Comment by kha84** [comment]
I recently came across a wierd issue with my several different headscale installations. I couldn't attach any new nodes to it:
- on a client side I was running `tailscale up --login-server https://myserver:444` (yes, I'm running it on a non-standard port)
- `tailscale` was printing out the URL to follow like `https://myserver:444/register/xxxx`
- but the built-in headscale web server wasn't responding on that URL (even after a restart)  
- I even tried to manually accept the node registration by running `headscale nodes register --user some-new-user --key xxxx`, but it was giving me an error `could not register node: node not found in registration cache` (again, the restart of headscale didn't help) 

I eventually managed **to work this around** by adding a node with using `preauthkey`, but I totally get @paladincorners pain - having no way to see what keys are currently in the cache of headscale is limiting debugging / troubleshooting options. It would be great, if dev team could add this feature.
